x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingResourceTracker.java

```diff
@@ -230,9 +272,86 @@ static void getMemoryAndCpu(
                 extraSingleNodeProcessors,
                 extraModelMemoryInBytes,
                 extraProcessors,
-                0,
+                removeNodeMemoryInBytes,
                 MachineLearning.NATIVE_EXECUTABLE_CODE_OVERHEAD.getBytes()
             )
         );
     }
+
+    /**
+     * Try to remove node memory
+     *
+     * @param perNodeUsedModelMemoryInBytes per Node lists of used model memory
+     * @param perNodeMemoryInBytes total model memory available on every node
+     * @return true if a node can be removed, false if not
+     */
+    static boolean tryRemoveOneNode(Map<String, List<Long>> perNodeUsedModelMemoryInBytes, long perNodeMemoryInBytes) {
+        if (perNodeUsedModelMemoryInBytes.size() <= 1) {
+            return false;
+        }
+
+        Map<String, Long> perNodeModelMemoryInBytesSum = perNodeUsedModelMemoryInBytes.entrySet()
+            .stream()
+            .map(entry -> tuple(entry.getKey(), entry.getValue().stream().mapToLong(Long::longValue).sum()))
+            .collect(Collectors.toMap(Tuple::v1, Tuple::v2));
+
+        Optional<Map.Entry<String, Long>> leastLoadedNodeAndMemoryUsage = perNodeModelMemoryInBytesSum.entrySet()
+            .stream()
+            .min(Comparator.comparingLong(Map.Entry::getValue));
+
+        if (leastLoadedNodeAndMemoryUsage.isPresent() == false) {
+            return false;
+        }
+
+        // this currently only works based on memory, not CPU
+        String candidateNode = leastLoadedNodeAndMemoryUsage.get().getKey();
+        List<Long> candidateNodeMemoryList = perNodeUsedModelMemoryInBytes.get(candidateNode);
+        perNodeModelMemoryInBytesSum.remove(candidateNode);
+
+        // if all jobs fit on other nodes, we can scale down one node
+        return tryMoveJobsByMemoryInLeastEfficientWay(candidateNodeMemoryList, perNodeModelMemoryInBytesSum, perNodeMemoryInBytes) == 0L;
+    }
+
+    /**
+     * Try to place jobs by size to other nodes in the leas efficient way
+     *
+     * @param candidateNodeMemoryList list of jobmemory values given by the candidate job
+     * @param perNodeModelMemoryInBytesSum other nodes memory usage
+     * @param perNodeMemoryInBytes available memory per node
+     * @return remaining memory, that could not be placed on other nodes, 0L if all jobs got placed
+     */
+    static long tryMoveJobsByMemoryInLeastEfficientWay(
```

Comment:
Do you think it's worth trying to handle negative values here? I'm sure we probably have other problems if nodes are reporting negative memory usage haha.

nit: when I read `tryMoveJobsByMemoryInLeastEfficientWay` I thought it was actually going to doing reallocations across nodes, maybe `calculateJobMemoryOverflow` or `calculateUnallocatableJobMemoryAmount` 🤷‍♂️ 

Also just curious why this is the least efficient way?

Reply:
added comments to explain. We basically have no influence on how jobs are placed. In case we recommend a downscale, we want to be sure not to scale up immediately if due to inefficient placement jobs are unassigned.